src/workerd/jsg/setup.h

```diff
@@ -541,59 +559,74 @@ class Isolate: public IsolateBase {
             returningFunction) override {
-      return jsgIsolate.wrapper->wrap(context, kj::none, kj::mv(returningFunction));
+      return jsgIsolate.getWrapperByContext(context)->wrap(
+          context, kj::none, kj::mv(returningFunction));
     }
     v8::Local<v8::Function> wrapPromiseReturningFunction(v8::Local<v8::Context> context,
         jsg::Function<jsg::Promise<jsg::Value>(const v8::FunctionCallbackInfo<v8::Value>& info)>
             returningFunction) override {
-      return jsgIsolate.wrapper->wrap(context, kj::none, kj::mv(returningFunction));
+      return jsgIsolate.getWrapperByContext(context)->wrap(
+          context, kj::none, kj::mv(returningFunction));
     }
     kj::String toString(v8::Local<v8::Value> value) override {
-      return jsgIsolate.wrapper->template unwrap<kj::String>(
+      return jsgIsolate.getWrapperByContext(*this)->template unwrap<kj::String>(
           v8Isolate->GetCurrentContext(), value, jsg::TypeErrorContext::other());
     }
     jsg::Dict<v8::Local<v8::Value>> toDict(v8::Local<v8::Value> value) override {
-      return jsgIsolate.wrapper->template unwrap<jsg::Dict<v8::Local<v8::Value>>>(
-          v8Isolate->GetCurrentContext(), value, jsg::TypeErrorContext::other());
+      return jsgIsolate.getWrapperByContext(*this)
+          ->template unwrap<jsg::Dict<v8::Local<v8::Value>>>(
+              v8Isolate->GetCurrentContext(), value, jsg::TypeErrorContext::other());
     }
     jsg::Dict<jsg::JsValue> toDict(const jsg::JsValue& value) override {
-      return jsgIsolate.wrapper->template unwrap<jsg::Dict<jsg::JsValue>>(
+      return jsgIsolate.getWrapperByContext(*this)->template unwrap<jsg::Dict<jsg::JsValue>>(
           v8Isolate->GetCurrentContext(), value, jsg::TypeErrorContext::other());
     }
     v8::Local<v8::Promise> wrapSimplePromise(jsg::Promise<jsg::Value> promise) override {
-      return jsgIsolate.wrapper->wrap(v8Context(), kj::none, kj::mv(promise));
+      return jsgIsolate.getWrapperByContext(*this)->wrap(v8Context(), kj::none, kj::mv(promise));
     }
     jsg::Promise<jsg::Value> toPromise(v8::Local<v8::Value> promise) override {
-      return jsgIsolate.wrapper->template unwrap<jsg::Promise<jsg::Value>>(
+      return jsgIsolate.getWrapperByContext(*this)->template unwrap<jsg::Promise<jsg::Value>>(
           v8Isolate->GetCurrentContext(), promise, jsg::TypeErrorContext::other());
     }
 
-    // Creates a new JavaScript "context", i.e. the global object. This is the first step to
-    // executing JavaScript code. T should be one of your API types which you want to use as the
-    // global object. `args...` are passed to the type's constructor.
     template <typename T, typename... Args>
-    JsContext<T> newContext(NewContextOptions options, Args&&... args) {
+    JsContext<T> newContextWithWrapper(
+        kj::Own<TypeWrapper>& wrapper, NewContextOptions options, Args&&... args) {
```

Comment:
Instead of passing `kj::Own<TypeWrapper>&`, why not just pass `TypeWrapper&`? It doesn't look like you actually do anything with the `Own` except deference it.

Reply:
I can pass a `TypeWrapper*` instead but I dislike the idea of passing a `Typewrapper&` and then getting it's address with `&wrapper`

Reply:
> I dislike the idea of passing a Typewrapper& and then getting it's address with &wrapper

Why? We do that all the time. I don't see the issue.